authors/apps/authentication/serializers.py

```diff
@@ -4,26 +4,51 @@
 
 from .models import User
 
+import re
+
 
 class RegistrationSerializer(serializers.ModelSerializer):
     """Serializers registration requests and creates a new user."""
 
     # Ensure passwords are at least 8 characters long, no longer than 128
     # characters, and can not be read by the client.
-    password = serializers.CharField(
-        max_length=128,
-        min_length=8,
-        write_only=True
-    )
+    password = serializers.CharField(max_length=128, min_length=8, write_only=True)
 
     # The client should not be able to send a token along with a registration
     # request. Making `token` read-only handles that for us.
 
+    def reg_fields_validation(self, username, email, password):
+        # if email is None:
+        #     raise serializers.ValidationError("An email address is required")
+
+        if re.compile("[!@#$%^&*:;?><.0-9]").match(username):
+            raise serializers.ValidationError(
+                "Invalid Username , it contains invalid characters."
+            )
+
+        if str(password).isdigit() :
+            raise serializers.ValidationError(
+                "Your password can not be numbers only!"
+            )
+
+        if re.compile("[!@#$%^&*:;?><.0-9]").search(password) is None:
+            raise serializers.ValidationError(
+                "Ensure your password is alphanumeric"
+            )
+
+    def validate(self, data):
+        username = data.get("username", None)
+        email = data.get("email", None)
+        password = data.get("password", None)
+
+        self.reg_fields_validation(username, email, password)
+        return {"email": email, "username": username, "password": password}
+
     class Meta:
         model = User
         # List all of the fields that could possibly be included in a request
         # or response, including fields specified explicitly above.
-        fields = ['email', 'username', 'password']
+        fields = ["email", "username", "password"]
```

Comment:
Why did you change to double quotes?

Reply:
@RachaelNantale this was done by pep8 formatting. since we resolved to use single quotes, this file has been updated to have single quotes.